aws/resource_aws_sagemaker_device_fleet.go

```diff
@@ -0,0 +1,252 @@
+		input.RoleArn = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if len(tags) > 0 {
+		input.Tags = tags.IgnoreAws().SagemakerTags()
+	}
+
+	_, err := retryOnAwsCode("ValidationException", func() (interface{}, error) {
+		return conn.CreateDeviceFleet(input)
+	})
+	if err != nil {
+		return fmt.Errorf("error creating SageMaker Device Fleet %s: %w", name, err)
+	}
+
+	d.SetId(name)
+
+	return resourceAwsSagemakerDeviceFleetRead(d, meta)
+}
+
+func resourceAwsSagemakerDeviceFleetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	deviceFleet, err := finder.DeviceFleetByName(conn, d.Id())
+	if err != nil {
+		if isAWSErr(err, "ValidationException", "No devicefleet with name") {
+			d.SetId("")
+			log.Printf("[WARN] Unable to find SageMaker Device Fleet (%s); removing from state", d.Id())
+			return nil
+		}
+		return fmt.Errorf("error reading SageMaker Device Fleet (%s): %w", d.Id(), err)
+
+	}
+
+	arn := aws.StringValue(deviceFleet.DeviceFleetArn)
+	d.Set("device_fleet_name", deviceFleet.DeviceFleetName)
+	d.Set("arn", arn)
+	d.Set("role_arn", deviceFleet.RoleArn)
+	d.Set("description", deviceFleet.Description)
+
+	iotAlias := aws.StringValue(deviceFleet.IotRoleAlias)
+	d.Set("iot_role_alias", iotAlias)
+	d.Set("enable_iot_role_alias", len(iotAlias) > 0)
+
+	if err := d.Set("output_config", flattenSagemakerFeatureDeviceFleetOutputConfig(deviceFleet.OutputConfig)); err != nil {
+		return fmt.Errorf("error setting output_config for Sagemaker Device Fleet (%s): %w", d.Id(), err)
+	}
+
+	tags, err := keyvaluetags.SagemakerListTags(conn, arn)
+
+	if err != nil {
+		return fmt.Errorf("error listing tags for SageMaker Device Fleet (%s): %w", d.Id(), err)
+	}
+
+	tags = tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig)
+
+	//lintignore:AWSR002
+	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	if err := d.Set("tags_all", tags.Map()); err != nil {
+		return fmt.Errorf("error setting tags_all: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAwsSagemakerDeviceFleetUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	if d.HasChangeExcept("tags_all") {
+		input := &sagemaker.UpdateDeviceFleetInput{
+			DeviceFleetName:    aws.String(d.Id()),
+			EnableIotRoleAlias: aws.Bool(d.Get("enable_iot_role_alias").(bool)),
+			OutputConfig:       expandSagemakerFeatureDeviceFleetOutputConfig(d.Get("output_config").([]interface{})),
+			RoleArn:            aws.String(d.Get("role_arn").(string)),
+		}
+
+		if d.HasChange("description") {
+			input.Description = aws.String(d.Get("description").(string))
+		}
+
+		log.Printf("[DEBUG] sagemaker DeviceFleet update config: %#v", *input)
```

Comment:
Since `UpdateDeviceFleetInput` defines a `String()` function, we can use the `%s` format verb here

Reply:
added